contractcourt/commit_sweep_resolver.go

```diff
@@ -252,18 +257,26 @@ func (c *commitSweepResolver) Resolve(_ bool) (ContractResolver, error) {
 
 		signDesc = c.commitResolution.SelfOutputSignDesc
 	)
+
 	switch {
 	// For taproot channels, we'll know if this is the local commit based
-	// on the witness script. For local channels, the witness script has an
-	// OP_DROP value.
-	//
-	// TODO(roasbeef): revisit this after the script changes
-	//  * otherwise need to base off the key in script or the CSV value
-	//  (script num encode)
+	// on the timelock value. For remote commitment transactions, the
+	// witness script has a timelock of 1.
 	case c.chanType.IsTaproot():
-		scriptLen := len(signDesc.WitnessScript)
-		isLocalCommitTx = signDesc.WitnessScript[scriptLen-1] ==
-			txscript.OP_DROP
+		delayKey := *c.localChanCfg.DelayBasePoint.PubKey
+		nonDelayKey := *c.localChanCfg.PaymentBasePoint.PubKey
+
+		signKey := *c.commitResolution.SelfOutputSignDesc.KeyDesc.PubKey
+
+		// If the key in the script is neither of these, we shouldn't
+		// proceed. This should be impossible.
+		if signKey != delayKey && signKey != nonDelayKey {
```

Comment:
should we use `IsEqual` instead? Not sure how this equality check `!=` behaves.

Reply:
Hmm, so they aren't pointers at this point, and the internal field are fully static: https://github.com/decred/dcrd/blob/141385ae9ea27f22013b1a5c0f63b0515c15c508/dcrec/secp256k1/pubkey.go#L65-L72

https://github.com/decred/dcrd/blob/141385ae9ea27f22013b1a5c0f63b0515c15c508/dcrec/secp256k1/field.go#L143-L172

With that said, one advantage of `IsEqual`, is that i'll use a constant time check: https://github.com/decred/dcrd/blob/141385ae9ea27f22013b1a5c0f63b0515c15c508/dcrec/secp256k1/field.go#L613-L628

Don't think we actually need such behavior in this case though.

Reply:
We shouldn't need to consider timing issues here because it is non-interactive wrt a potential adversary and these are pubkey comparisons anyway. Happy to use `IsEqual` if that is the proper way to do it, but I'm not sure I understand the reason at this point.